src/Expr.cc

```diff
@@ -4747,32 +4751,28 @@ LambdaExpr::LambdaExpr(std::unique_ptr<function_ingredients> arg_ing, IDPList ar
 	lambda_id->SetConst();
 	}
 
-void LambdaExpr::CheckCaptures(StmtPtr when_parent)
+bool LambdaExpr::CheckCaptures(StmtPtr when_parent)
 	{
 	auto ft = type->AsFuncType();
 	const auto& captures = ft->GetCaptures();
 
-	capture_by_ref = false;
+	auto desc = when_parent ? "\"when\" statement" : "lambda";
 
 	if ( ! captures )
 		{
 		if ( outer_ids.size() > 0 )
```

Comment:
```suggestion
		if ( ! outer_ids.empty() )
```

Reply:
Oops yeah, thx.  I try to canonicalize those but sometimes miss some!